src/lib/registration/connect_helpers.rb

```diff
@@ -96,8 +103,12 @@ def self.catch_registration_errors(message_prefix: "", show_update_hint: false,
           check_smt_api(error_msg)
           report_error(message_prefix + _("Connection to registration server failed."), error_msg)
         when 422
-          # Error popup
-          report_error(message_prefix + _("Connection to registration server failed."), error_msg)
+          if silent_reg_code_mismatch && e.message =~ /does not include the requested product/
```

Comment:
Be careful, the `message` text is a *translated* error message, this won't work in non-English locale.

See https://github.com/SUSE/connect/blob/master/lib/suse/connect/errors.rb#L35

We should use `e.response.body["error"]` instead.